src/components/Cart/Cart.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useCallback, useEffect } from 'react';
+import {
+  useDispatch, useSelector,
+} from 'react-redux';
+import {
+  Row, Col,
+} from 'react-bootstrap';
+import { useParams, useLocation, useHistory } from 'react-router';
+import queryString from 'query-string';
+
+import { addToCart, removeFromCart } from '../../redux/actions/cartActions';
+import ItemsList from './ItemsList/ItemsList';
+import CheckoutCard from './CheckoutCard/CheckoutCard';
+
+const Cart = () => {
+  const { id: productId } = useParams();
+  const history = useHistory();
+  const { search } = useLocation();
+  const dispatch = useDispatch();
+  const { cartItems } = useSelector((state) => state.cart);
+
+  const { qty } = queryString.parse(search);
+
+  useEffect(() => {
+    if (productId) {
+      dispatch(addToCart(productId, Number(qty)));
+    }
+  }, [dispatch, productId, qty]);
+
+  const handleQtyChange = useCallback((id) => ({ target: { value } }) => {
```

Comment:
why do u wrap all the functions with useCallback()?!

Reply:
cause it's better for the performance as it doesn't redcleared within every render